akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMultiMap.scala

```diff
@@ -41,27 +48,35 @@ object ORMultiMap {
  * This class is immutable, i.e. "modifying" methods return a new instance.
  */
 @SerialVersionUID(1L)
-final class ORMultiMap[A, B] private[akka] (private[akka] val underlying: ORMap[A, ORSet[B]])
-  extends ReplicatedData with ReplicatedDataSerialization with RemovedNodePruning {
+final class ORMultiMap[A, B] private[akka] (
+  private[akka] val underlying: ORMap[A, ORSet[B]],
+  private[akka] val mapType:    Int)
+  extends DeltaReplicatedData with ReplicatedDataSerialization with RemovedNodePruning {
 
   override type T = ORMultiMap[A, B]
+  override type D = ORMap.DeltaOp
 
   override def merge(that: T): T =
-    new ORMultiMap(underlying.merge(that.underlying))
+    if (that.mapType == mapType) {
+      if (mapType == ORMultiMap.DeltaDeltaORMultiMapTag)
+        new ORMultiMap(underlying.mergeRetainingDeletedValues(that.underlying), mapType)
+      else
+        new ORMultiMap(underlying.merge(that.underlying), mapType)
+    } else throw new IllegalArgumentException("Trying to merge two ORMultiMaps of different mapType")
 
   /**
    * Scala API: All entries of a multimap where keys are strings and values are sets.
    */
   def entries: Map[A, Set[B]] =
-    underlying.entries.map { case (k, v) ⇒ k → v.elements }
+    underlying.entries.filter { case (k, _) ⇒ underlying.keys.elements.contains(k) }.map { case (k, v) ⇒ k → v.elements }
```

Comment:
that's only needed if using delta-values, right?
if so, make it an `if`

use `collect` instead of `filter + map`